cmd/soroban-rpc/internal/db/ledgerentry.go

```diff
@@ -253,7 +253,7 @@ func (l *ledgerEntryReadTx) GetLedgerEntry(key xdr.LedgerKey, includeExpired boo
 			if err != nil {
 				return false, xdr.LedgerEntry{}, err
 			}
-			if expirationLedgerSeq <= xdr.Uint32(latestClosedLedger) {
+			if expirationLedgerSeq < xdr.Uint32(latestClosedLedger) {
```

Comment:
Wouldn't this give an extra ledger of visibility? The latestClosedLedger is the one *before* the current ledger we're building now.

Reply:
That’s just how state expiration is defined at https://soroban.stellar.org/docs/fundamentals-and-concepts/state-expiration

Reply:
More specifically 

The entry is considered expired when current_ledger > expirationLedger.

Reply:
FWIW if the entry expires on `latestClosedLedger + 1` (i.e. it's `expirationLedgerSeq == latestClosedLedger`), then the transaction you preflight should see the entry as expired. The same goes about retrieving contract data  - while the entry has been alive for LCL, it might already be expired at the moment we fetch it.